cirq/experiments/cross_entropy_benchmarking.py

```diff
@@ -0,0 +1,363 @@
+
+        q10    q11 ── q12    q13            q10    q11    q12    q13
+                                             |      |      |      |
+        q20    q21 ── q22    q23            q20    q21    q22    q23
+
+        q30    q31 ── q32    q33            q30    q31    q32    q33
+
+    Args:
+        qubits: The grid qubits included in the entangling operations.
+        two_qubit_gate: The two-qubit gate to be applied between all
+            neighboring pairs of qubits.
+
+    Returns:
+        A list of sets of operations, with a maximum length of 4. Each set
+        includes two-qubit gates which can be performed at the same time.
+    """
+    interaction_sequence = _default_interaction_sequence(qubits)
+    return [{two_qubit_gate(q_a, q_b) for (q_a, q_b) in pairs} for pairs in
+            interaction_sequence]
+
+
+def _build_xeb_circuits(qubits: Sequence[ops.Qid],
+                        cycles: Sequence[int],
+                        single_qubit_gates: List[List[
+                            ops.SingleQubitGate]] = None,
+                        benchmark_ops: Sequence[Set[ops.OP_TREE]] = None,
+                        ) -> List[circuits.Circuit]:
+    if benchmark_ops is not None:
+        num_d = len(benchmark_ops)
+    else:
+        num_d = 0
+    max_cycles = max(cycles)
+
+    if single_qubit_gates is None:
+        single_rots = _random_half_rotations(qubits, max_cycles)
+    else:
+        single_rots = _random_any_gates(qubits, single_qubit_gates, max_cycles)
+    all_circuits = []  # type: List[circuits.Circuit]
+    for num_cycles in cycles:
+        circuit_exp = circuits.Circuit()
+        for i in range(num_cycles):
+            circuit_exp.append(single_rots[i])
+            if num_d > 0:
+                for op_set in benchmark_ops[i % num_d]:
+                    circuit_exp.append(op_set)
+        all_circuits.append(circuit_exp)
+    return all_circuits
+
+
+def _measure_prob_distribution(sampler: sim.Sampler, repetitions: int,
+                               qubits: Sequence[ops.Qid],
+                               circuit_list: List[circuits.Circuit]
+                               ) -> List[np.ndarray]:
+    all_probs = []  # type: List[np.ndarray]
+    num_states = 2 ** len(qubits)
+    for circuit_i in circuit_list:
+        trial_circuit = circuit_i.copy()
+        trial_circuit.append(ops.measure(*qubits, key='z'))
+        res = sampler.run(trial_circuit, repetitions=repetitions)
+        res_hist = dict(res.histogram(key='z'))
+        probs = np.zeros(num_states, dtype=float)
+        for k, v in res_hist.items():
+            probs[k] = float(v) / float(repetitions)
+        all_probs.append(probs)
+    return all_probs
+
+
+def _xeb_fidelities(ideal_probs: Dict[int, np.ndarray],
+                    actual_probs: Dict[int, np.ndarray]) -> List[float]:
+    num_cycles = sorted(list(ideal_probs.keys()))
+    xeb_fidelity = []  # type: List[float]
```

Comment:
This can be replaced by a simple list comprehension.

Reply:
Yes, changed the code here and now it is shorter.